components/03-modules/filter/_filter.scss

```diff
@@ -0,0 +1,144 @@
+$filter__swatch-option-margin              : $spacer--medium !default;
+$filter__swatch-option-background          : $white !default;
+$filter__swatch-option-color               : $gray-dark !default;
+$filter__swatch-option-border              : 3px solid $white !default;
+$filter__swatch-option-border-color-hover  : $color-primary !default;
+$filter__swatch-option-border-color--active: $color-primary !default;
+// Label
+$filter__label-padding                     : $spacer--medium 0 !default;
+// Color
+$filter__color-margin                      : 0 0 $spacer--medium 0 !default;
+$filter__color-option-width                : 56px !default;
+$filter__color-option-height               : 80px !default;
+$filter__color-option-after-top            : 4px !default;
+$filter__color-option-after-left           : 8px !default;
+$filter__color-option-after-size           : 40px !default;
+$filter__color-option-after-border-radius  : 100% !default;
+$filter__color-option-after-border         : 2px solid $color-primary !default;
+$filter__color-sample-size                 : 32px !default;
+$filter__color-sample-margin               : $spacer !default;
+$filter__color-sample-border-radius        : 100% !default;
+$filter__color-label-font-size             : 10px !default;
+$filter__color-label-color                 : $gray-darkest !default;
+// Show-all
+$filter__show-all-height                   : 48px !default;
+$filter__show-all-font-size                : 12px !default;
+// Divider
+$filter__divider-border-top                : solid 1px lighten($gray-lighter, 15%) !default;
+// Range
+$filter__range-margin                      : 0 0 $spacer--medium 0 !default;
+
+.filter {
+    &__label {
+        padding: $filter__label-padding;
+    }
+
+    &__swatch {
+        display: flex;
+        flex-wrap: wrap;
+        margin: $filter__swatch-margin;
+        &-option {
+            display: flex;
+            justify-content: center;
+            align-items: center;
+            min-width: $filter__swatch-option-size;
+            min-height: $filter__swatch-option-size;
+            margin-right: $filter__swatch-option-margin;
+            border: $filter__swatch-option-border;
+            background-color: $filter__swatch-option-background;
+            color: $filter__swatch-option-color;
+            transition: $filter__transition;
+            cursor: pointer;
+            text-decoration: none;
+
+            &:hover,
+            &:focus {
+                text-decoration: none;
+                border-color: $filter__swatch-option-border-color-hover;
+            }
+
+            &:last-child {
+                margin-right: 0;
+            }
+
+            &--active {
+                border-color: $filter__swatch-option-border-color--active;
+            }
+        }
+    }
+
+    &__color {
+        display: flex;
+        flex-flow: row wrap;
+        margin: $filter__color-margin;
+
+        &-option {
```

Comment:
the same thing about mixing element with element in scss structure

Reply:
Done